### ISSUE TITLE
CleanMX Parser - Dumping message

### ISSUE THREAD

**Issue by SYNchroACK** [certtools/intelmq#1135]
CleanMX found one error regarding a bad domain... (which is something normal... )
```
2017-12-12 09:32:09,402 - cleanmx-phishing-parser - INFO - Processed 500 messages since last logging.
2017-12-12 09:32:09,746 - cleanmx-phishing-parser - ERROR - Failed to parse line.
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/intelmq/lib/bot.py", line 616, in process
    events = list(filter(bool, self.parse_line(line, report)))
  File "/usr/local/lib/python3.5/dist-packages/intelmq/bots/parsers/cleanmx/parser.py", line 144, in parse_line
    event.add(key, value)
  File "/usr/local/lib/python3.5/dist-packages/intelmq/lib/message.py", line 223, in add
    raise exceptions.InvalidValue(key, old_value)
intelmq.lib.exceptions.InvalidValue: invalid value 'accounts.google.' (<class 'str'>) for key 'source.ip'
```

And in the end, when tries to dump the message encounter another problem:
```
2017-12-12 09:32:42,498 - cleanmx-phishing-parser - INFO - Processed 500 messages since last logging.
2017-12-12 09:32:42,717 - cleanmx-phishing-parser - ERROR - Bot has found a problem.
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/intelmq/lib/bot.py", line 144, in start
    self.process()
  File "/usr/local/lib/python3.5/dist-packages/intelmq/lib/bot.py", line 625, in process
    report_dump.change('raw', self.recover_line(line))
  File "/usr/local/lib/python3.5/dist-packages/intelmq/lib/bot.py", line 636, in recover_line
    return '\n'.join(self.tempdata + [line])
TypeError: sequence item 0: expected str instance, dict found
2017-12-12 09:32:42,718 - cleanmx-phishing-parser - INFO - Current Message(event): {'feed.provider': 'CleanMX', 'raw': 'ImxpbmUiLCJpZCIsImZpcnN0IiwiZmlyc3R0aW1lIiwibGFzdCIsImxhc3R0aW1lIiwicGhpc2h0YW5rIiwidmlydXNuYW1lIiwidXJsIiwicmV
jZW50IiwicmVzcG9uc2UiLCJpcCIsInJldmlldyIsImRvbWFpbiIsImNvdW50cnkiLCJzb3VyY2UiLCJlbWFpbCIsImluZXRudW0iLCJuZXRuYW1lIiwiZGRlc2NyIiwibnMxIiwibnMyIiwibnMzIiwibnM0IiwibnM1IgoiMSIsIjExMDk1Mjc1IiwiMTUwMzM1MjEyMSIsIjIwMTctMDgtMjEgMjM6NDg6ND
EiLCIwIiwiMTk3MC0wMS0wMSAwMTowMDowMCIsIjUxNzIyNDQiLCIiLCJodHRwOi8va2hhbmhnaWFuZy5jb20vcHVibGljL2VkaXRvci9zb3VyY2UvZGZkd3Nkai9sb2xzZW ....
```

**Comment by wagner-certat** [comment]
Which version/branch/revision are you using?

**Comment by wagner-certat** [comment]
Could you test if adding `recover_line = ParserBot.recover_line_csv_dict` to the `CleanMXParserBot` class does fix this bug?

**Comment by SYNchroACK** [comment]
related #1136

**Comment by wagner-certat** [comment]
Is #1136 the bug fix for this?

**Comment by SYNchroACK** [comment]
> Is #1136 the bug fix for this?

yes. These errors were consequences of bad data from CSV format. Now with XML I expect to not have the these issues.

**Comment by wagner-certat** [comment]
As #1136 is still missing a changelog entry, please add the issue number there and also write `fixes certtools/intelmq#1135` into the commit message, thanks!
